src/components/Registration.jsx

```diff
@@ -0,0 +1,111 @@
+import {
+  Card
+} from "@mui/material";
+import React, { useState, useEffect } from "react";
+import { StyledCardContent, StyledButton, StyledForm, StyledTitle, Heading, StyledTextField } from '../styles/styled';
+
+export const Registration = ({ onRegistration }) => {
+  const [user, setUser] = useState({
+    username: "",
+    email: "",
+    password: "",
+    passwordConfirm: "",
+  });
+
+  const [isValidForm, setIsValidForm] = useState(false);
+
+  useEffect(() => {
+    if (validateEmail(user.email) && user.password && onConfirmPassword()) {
+      setIsValidForm(true);
+    } else {
+      setIsValidForm(false);
+    }
+  }, [user]);
+
+  const validateEmail = (email) => {
+    return /^\S+@\S+\.\S+$/.test(email);
+  };
+
+  const extractUsername = (email) => {
+    return email.slice(0, email.indexOf("@"));
+  };
+
+  const onChangeEmail = (e) => {
+    if (validateEmail(e.target.value)) {
+      setUser({
+        ...user,
+        username: extractUsername(e.target.value),
+        email: e.target.value,
+      });
+    } else {
+      setUser({
+        ...user,
+        username: "",
+        email: e.target.value,
+      });
+    }
+  };
+
+  const onConfirmPassword = () => {
+    return user.password === user.passwordConfirm;
+  };
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+    console.log(user);
+    onRegistration();
+  };
+
+  return (
+    <Card>
+      <StyledCardContent>
+        <StyledTitle>
+          <Heading>REGISTRATION</Heading>
+        </StyledTitle>
+        <StyledForm onSubmit={onSubmit}>
+          <StyledTextField
+            label="Email Address"
+            value={user.email}
+            onChange={onChangeEmail}
+            error={Boolean(user.email) && !validateEmail(user.email)}
+            helperText={
+              user.email && !validateEmail(user.email)
+                ? "Incorrect format of Email Address!"
+                : ""
+            }
+          />
+          <StyledTextField
+            label="User Name"
+            value={user.username}
+            inputProps={{ readOnly: true }}
```

Comment:
Tohle je dobrý technický řešení toho jak nepřepisovat uživateli username, ale není to vůbec real-world ready a UX friendly. Určitě chceš uživateli dovolit si uživatelské jméno změnit, např může existovat uživatel s emailem franta@gmail.com a franta@seznam.com. Ti 2 by měli stejné uživatelské jméno. Pak záleží jestli chceš aby jméno bylo unikátní nebo ne ale dá se ještě vygenerovat nějaký číslo za něj, třeba druhý uživatel by měl franta1.

To je ale už dost advanced téma a obnášelo by databáze, logiku systému atd. Tohle řešení funguje a je dobré v rámci úkolu :) 
Jediný co by se dalo ještě zlepšit je nějak vizuálně dát uživateli vědět že to políčko nejde vyplnit. Takhle vypadá stejně jako ostatní políčka a že do něj nejde psát se uživatel dozví až když zkusí kliknout a psát. Ideálně chceš aby to bylo jasný ještě před tím než to někdo bude chtít zkusit (=dobré UX)